js/data/data_source/data_source.js

```diff
@@ -205,11 +205,21 @@ var DataSource = Class.inherit({
         * @type Enums.DataSourceStoreType
         */
 
+        /**
+        * @name DataSourceOptions.pushAggregationTimeout
+        * @type number
+        * @default undefined
+        */
+        this._pushThrottle = new dataUtils.ThrottleWithAggregation(this._onPush, options.pushAggregationTimeout);
```

Comment:
This object will be created for each widget with dataSource even if it does not use push functionality. This is not good.